src/textfield/textfield.js

```diff
@@ -132,7 +132,9 @@
     if (this.input_.validity.valid) {
       this.element_.classList.remove(this.CssClasses_.IS_INVALID);
     } else {
-      this.element_.classList.add(this.CssClasses_.IS_INVALID);
+      if (this.element_.getElementsByTagName('input')[0].value.length > 0) {
```

Comment:
As you can see [here](https://github.com/rzr999/material-design-lite/blob/b478b00ddf4329c2a6a592dcfe6c6a0886ec04f4/src/textfield/textfield.js#L195), there is no need to requery for the input field. Please use `this.input_`.